insights/parsers/tests/test_lscpu.py

```diff
@@ -0,0 +1,101 @@
+import doctest
+import pytest
+from insights.parsers import lscpu, SkipException
+from insights.tests import context_wrap
+
+
+LSCPU_1 = """
+Architecture:          x86_64
+CPU op-mode(s):        32-bit, 64-bit
+Byte Order:            Little Endian
+CPU(s):                2
+On-line CPU(s) list:   0,1
+Thread(s) per core:    2
+Core(s) per socket:    1
+Socket(s):             1
+NUMA node(s):          1
+Vendor ID:             GenuineIntel
+CPU family:            6
+Model:                 60
+Model name:            Intel Core Processor (Haswell, no TSX)
+Stepping:              1
+CPU MHz:               2793.530
+BogoMIPS:              5587.06
+Hypervisor vendor:     KVM
+Virtualization type:   full
+L1d cache:             32K
+L1i cache:             32K
+L2 cache:              4096K
+NUMA node0 CPU(s):     0,1
+""".strip()
+
+LSCPU_2 = """
+Architecture:          x86_64
+CPU op-mode(s):        32-bit, 64-bit
+Byte Order:            Little Endian
+CPU(s):                2
+On-line CPU(s) list:   0
+Off-line CPU(s) list:  1
+Thread(s) per core:    1
+Core(s) per socket:    1
+Socket(s):             1
+NUMA node(s):          1
+Vendor ID:             GenuineIntel
+CPU family:            6
+Model:                 60
+Model name:            Intel Core Processor (Haswell, no TSX)
+Stepping:              1
+CPU MHz:               2793.530
+BogoMIPS:              5587.06
+Hypervisor vendor:     KVM
+Virtualization type:   full
+L1d cache:             32K
+L1i cache:             32K
+L2 cache:              4096K
+NUMA node0 CPU(s):     0
+""".strip()
+
+BLANK = """
+""".strip()
+
+BAD_LSCPU = """
+Architecture:          x86_64
+CPU op-mode(s) =        32-bit, 64-bit
```

Comment:
For this kind of abnormal lines, I think it's necessary to raise a `ParserExeption`, but not skip it.

Reply:
AFAIK `ParserException` will skip all the data where in this case the valid key-values will be captured. Finally the example used `=` will not ever happen in the output of `lscpu`. The reason I used the example to have full code coverage.

Reply:
Since it will never happen, lease remove this line

Reply:
@xiangce It will reduce the coverage from 100% to 96%. Do you really want me to remove it?

Reply:
As per the following code:
```python
            if split_on in line:
                k, v = line.split(split_on, 1)
                self.info[k.replace("(s)", "s").strip()] = v.strip()
``` 

 The [L68](https://github.com/RedHatInsights/insights-core/pull/2135/files#diff-1ab499a055ecdc312d029809c71dbd71R68) expects a `':'` in the line, and if there is not, the line will be skipped. And since there is such a check, it indicates some lines might not contain a `':'`.

Although, as you mentioned, this abnormal line ("xx= yy") will never happen, how about other cases?
If there are no other cases, why do we check it in L68? Since you have already thought there might be some lines missing the `':'`, and it's abnormal, it's reasonable to stop the parsing and raise an Exception to us.

But if you make sure there will be no lines missing the `':'`,  the L68 can be removed directly, right?

The goal of 100% coverage test is to help us to find any potential risk of the code. We'd better do all we can think of.

Reply:
@xiangce What is the issue with the current implementation? As per my understanding having an additional check won't hurt at all.

Reply:
@xiangce I removed L68(check for ":") as per your suggestion as the code was not good. Please check now.